### PR TITLE
Chimes option

### PR DESCRIPTION
**Pull request by SteveAmor**
Addresses #778

### CODE REVIEW

File: src/components/datetime/DateTimeController.cpp
Comment (Avamander) [inline src/components/datetime/DateTimeController.cpp:80]: Brace style inconsistent with the rest of the code

File: src/components/datetime/DateTimeController.cpp
Comment (Avamander) [inline src/components/datetime/DateTimeController.cpp:78]: Needs formatting with `clang-format`.

Conditional does not match the syntax used in the rest of the code.

**Review by Avamander — CHANGES_REQUESTED** [review]

File: src/components/datetime/DateTimeController.cpp
Comment (Avamander) [inline src/components/datetime/DateTimeController.cpp:89]: Lack of braces is inconsistent with the rest of the code.

File: src/components/datetime/DateTimeController.cpp
Comment (Avamander) [inline src/components/datetime/DateTimeController.cpp:86]: Usage of `and` and `not` is not consistent with the rest of the code.

**Review by Avamander — CHANGES_REQUESTED** [review]

File: src/components/datetime/DateTimeController.cpp
Comment (SteveAmor) [reply src/components/datetime/DateTimeController.cpp]: Hopefully my clang-format fixed this now?

File: src/components/datetime/DateTimeController.cpp
Comment (SteveAmor) [reply src/components/datetime/DateTimeController.cpp:89]: Now MD is causing me issues.  Is this what is required (from a braces point of view) to resolve please?

   if (systemTask != nullptr) {  
       systemTask->PushMessage(System::Messages::OnNewHalfHour);  
    }

File: src/components/datetime/DateTimeController.cpp
Comment (SteveAmor) [reply src/components/datetime/DateTimeController.cpp:86]: Is this preferred please?
`if ((minute == 0 or minute == 30) && !isHalfHourAlreadyNotified)`

Comment (Avamander) [reply src/components/datetime/DateTimeController.cpp:86]: Yep, that's good. Don't forget the closing brace though :D

File: src/components/datetime/DateTimeController.cpp
Comment (Avamander) [reply src/components/datetime/DateTimeController.cpp:89]: Yep, just adding the two, here and other places where they're missing.

File: src/components/datetime/DateTimeController.cpp
Comment (SteveAmor) [reply src/components/datetime/DateTimeController.cpp:86]: Should I also replace the `or` with `||` too?

Comment (Avamander) [reply src/components/datetime/DateTimeController.cpp:86]: That would be more consistent, yes.

**Comment by geekbozu** [comment]
Please update the settings version in the settings file for this set of changes.
 
https://github.com/InfiniTimeOrg/InfiniTime/blob/822b6ae6a0305c20a02a3e6e79f6f58296d585d4/src/components/settings/Settings.h#L176

This will prevent settings version conflicts.

**Review by Avamander — APPROVED** [review]

**Review by geekbozu — APPROVED** [review]
Looks great to me

File: src/displayapp/screens/settings/SettingChimes.cpp
Comment (NeroBurner) [inline src/displayapp/screens/settings/SettingChimes.cpp:1]:
```suggestion
#include "displayapp/screens/settings/SettingChimes.h"
```
could you make the include relative to the `src` directory as done in the (not yet merged, but approved) PR https://github.com/InfiniTimeOrg/InfiniTime/pull/805

**Review by NeroBurner — CHANGES_REQUESTED** [review]

**Comment by geekbozu** [comment]
So I've been using this for a while now, Id love something to differentiate it from a normal notification, Like more buzzes or a weird pattern or something. Might not be doable in this PR but some feedback. I'm always thinking I have a notification when it goes off :D

**Comment by SteveAmor** [comment]
> So I've been using this for a while now, Id love something to differentiate it from a normal notification, Like more buzzes or a weird pattern or something. Might not be doable in this PR but some feedback. I'm always thinking I have a notification when it goes off :D

I thought about a different notification method but didn't want to insert blocking code 'pulsing the motor a couple of times.

**Comment by lman0** [comment]
is this pull can be merged by @JF002   , @geekbozu ?
do you have other part to add @SteveAmor ?

**Comment by trman** [comment]
any news @SteveAmor @geekbozu ?
i do like this pr as it something i have activated in my ex peeble and wish as well to see here....

**Comment by geekbozu** [comment]
@SteveAmor if you could fix the now existing conflicts... :)

@lman0 @trman Once we approve them its up to @jf002 to merge them. All we can do is guide him. He will get there when his schedule allows and on his priority

**Comment by SteveAmor** [comment]
@geekbozu I have updated the conflicting files in commit https://github.com/InfiniTimeOrg/InfiniTime/pull/813/commits/94561b233136334c7507925b04a345adb0e6c769. I thought GitHub would realise and remove the above message that says This branch has conflicts that must be resolved but it is still there.

File: src/displayapp/screens/settings/SettingChimes.cpp
Comment (SteveAmor) [reply src/displayapp/screens/settings/SettingChimes.cpp]: updated in https://github.com/InfiniTimeOrg/InfiniTime/pull/813/commits/94561b233136334c7507925b04a345adb0e6c769

**Comment by Avamander** [comment]
@SteveAmor It's not GitHub, it's Git. There's unresolved conflicts still, make a merge commit or rebase.

File: src/displayapp/screens/settings/SettingChimes.cpp
Comment (SteveAmor) [reply src/displayapp/screens/settings/SettingChimes.cpp]: Done

File: src/components/settings/Settings.h
Comment (JF002) [inline src/components/settings/Settings.h:185]: I think the code would be clearer if you used a `enum class` instead of a plain integer : instead of comparing chimeState with 1, 2 or 3, you would compare it with `ChimeOptions::None`, `ChimeOptions::Hours`, `ChimeOptions::HalfHours`, for example

**Review by JF002 — CHANGES_REQUESTED** [review]
It looks pretty good. Could you have a look at the type of chimeState (enum classs instead of int) ?

File: src/components/settings/Settings.h
Comment (JF002) [reply src/components/settings/Settings.h]: Perfect! Thank you very much!

**Review by JF002 — APPROVED** [review]

**Pull request merged**
